WordPressKit/JSONDecoderExtension.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+extension JSONDecoder {
+
+    static var apiDecoder: JSONDecoder {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = JSONDecoder.DateDecodingStrategy.supportMultipleDateFormats
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        return decoder
+    }
+}
+
+extension JSONDecoder.DateDecodingStrategy {
+    
+    static var supportMultipleDateFormats: JSONDecoder.DateDecodingStrategy {
+        return JSONDecoder.DateDecodingStrategy.custom({ (decoder) -> Date in
+            let container = try decoder.singleValueContainer()
+            let dateStr = try container.decode(String.self)
+            let dateNoTimeFormatter = DateFormatter()
+            dateNoTimeFormatter.dateFormat = "yyyy-mm-dd"
+            
+            let dateWithTimeFormatter = DateFormatter()
+            dateWithTimeFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
```

Comment:
I don't _think_ having two `DateFormatter`s is necessary. Maybe something like this will work?
```
let dateFormatter = DateFormatter()
...
if len == 10 {
    dateFormatter.dateFormat = "yyyy-mm-dd"
    date = dateFormatter.date(from: dateStr)
} else if len == 19 {
    dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
    date = dateFormatter.date(from: dateStr)
}
```

Reply:
I refactored this